application/common/models/User.php

```diff
@@ -263,19 +263,26 @@ public static function findByEmail(string $email)
      */
     public function getAttributesForEmail()
     {
-        return [
+        $attrs = [
             'employeeId' => $this->employee_id,
             'firstName' => $this->first_name,
             'lastName' => $this->last_name,
-            'displayName' => $this->display_name,
+            'displayName' => $this->getDisplayName(),
             'username' => $this->username,
             'email' => $this->email,
             'active' => $this->active,
             'locked' => $this->locked,
             'lastChangedUtc' => $this->last_changed_utc,
             'lastSyncedUtc' => $this->last_synced_utc,
             'lastLoginUtc' => $this->last_login_utc,
+            'passwordExpiresUtc' => null, // Entry needed even if null.
+            'isMfaEnabled' => count($this->mfas) > 0 ? true : false,
         ];
+        if ($this->currentPassword !== null) {
+            $attrs['passwordExpiresUtc'] = $this->currentPassword->getGracePeriodEndsOn();
```

Comment:
Should we be using the expiration date, rather than the end of the grace period?

Reply:
yeah probably

Reply:
fixed